packages/vuetify/src/mixins/routable/index.ts

```diff
@@ -19,33 +23,71 @@ export default Vue.extend({
       default: undefined
     } as PropValidator<boolean | undefined>,
     exactActiveClass: String,
+    link: Boolean,
     href: [String, Object],
     to: [String, Object],
     nuxt: Boolean,
     replace: Boolean,
-    ripple: [Boolean, Object],
+    ripple: {
+      type: [Boolean, Object],
+      default: null
+    },
     tag: String,
     target: String
   },
 
+  data: () => ({
+    isActive: false,
+    proxyClass: ''
+  }),
+
   computed: {
+    classes (): object {
+      return {
+        [this.activeClass]: this.isActive,
+        [this.proxyClass]: this.isActive
+      }
+    },
     computedRipple (): RippleOptions | boolean {
-      return (this.ripple && !this.disabled) ? this.ripple : false
+      return this.ripple != null ? this.ripple : !this.disabled && this.isClickable
+    },
+    isClickable (): boolean {
+      if (this.disabled) return false
+
+      return Boolean(
+        this.isLink ||
+        this.$listeners.click ||
+        this.$listeners['!click'] ||
+        this.$attrs.tabindex
+      )
+    },
+    isLink (): boolean {
+      return this.to || this.href || this.link
+    },
+    styles (): object {
+      return {}
     }
   },
 
+  watch: {
+    $route: 'onRouteChange'
+  },
+
   methods: {
     click (e: MouseEvent): void {
       this.$emit('click', e)
     },
-    generateRouteLink (classes: object, styles: object = {}) {
+    generateRouteLink () {
       let exact = this.exact
       let tag
 
       const data: VNodeData = {
-        attrs: { disabled: this.disabled },
-        class: classes,
-        style: styles,
+        attrs: {
+          disabled: this.disabled,
+          tabindex: 'tabindex' in this.$attrs ? this.$attrs.tabindex : undefined
```

Comment:
`...this.$attrs`
or
`tabindex: this.$attrs.tabindex || undefined`
Please explain

Reply:
in the first scenario, you only want `tabindex` from `this.$attrs` not everything from it
in the second scenario, `this.$attrs.tabindex` will throw an error if `this` or `this.$attrs` is undefined
🤷‍♂